base_layer/core/src/blocks/block_header.rs

```diff
@@ -168,16 +169,34 @@ impl BlockHeader {
         }
     }
 
+    /// Returns the [std::cmp::Ordering] of the timestamp of this header compared to the "other" header
+    pub fn compare_timestamp(&self, other: &BlockHeader) -> Ordering {
+        self.timestamp.as_u64().cmp(&other.timestamp.as_u64())
```

Comment:
`EpochTime` has a `PartialOrd` impl that does this comparison.

Reply:
Whoops lol it did feel wrong